### ISSUE TITLE
Code quality: Remove / reduce functions and classes not defined at global scope

### ISSUE THREAD

**Issue by dhermes** [google/oauth2client#433]
Some examples:

- all the [`ndb` helpers][1] in the `appengine` module (changes coming)
- the class created in [`OAuth2Decorator.callback_handler`][2]
- version specific [Django config][3]
- locked file [`fcntl` helpers][4]
- locked file [`win32` helpers][5]
- class-within-a-class storage [in `multistore_file`][6]
- monkey-patched [request method][7] (this is a whole project in itself)

Also worth noting, but probably not worth changing are decorators:

- appengine decorators return values ([`oauth_required`][8] and [`oauth_aware`][9])
- [decorators][10] in `util`
- [decorators][11] in `django_util` ([also][12])
- [decorators][13] in `flask_util`

[1]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/appengine.py#L94
[2]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/appengine.py#L916
[3]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/django_util/apps.py#L29
[4]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/locked_file.py#L179
[5]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/locked_file.py#L248
[6]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/multistore_file.py#L213
[7]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/client.py#L592
[8]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/appengine.py#L762
[9]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/appengine.py#L829
[10]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/util.py#L120-L122
[11]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/django_util/decorators.py#L51-L53
[12]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/django_util/decorators.py#L102-L104
[13]: https://github.com/google/oauth2client/blob/349942b0db8e96c336802dc421fd4f1670e4b7e8/oauth2client/contrib/flask_util.py#L500-L502

**Comment by jonparrott** [comment]
What would you change about the decorators?

**Comment by jonparrott** [comment]
> monkey-patched request method (this is a whole project in itself)

Indeed. I'm doing some experiments to see how I can rip httplib2 out of this library. If you want to hold off on that until 100% coverage we can discuss a plan of attack.

**Comment by dhermes** [comment]
RE: "decorators", probably nothing, though maybe classes with `__call__` defined. I just included them since they came up in my `git grep` search.

RE: 100% coverage. We need a "coalition of volunteers" to get #212 over the hill.

RE: Ripping out `httplib2`, yay!

**Comment by jonparrott** [comment]
> RE: "decorators", probably nothing

Gotcha. I personally don't think decorator inner functions need to necessarily be pulled out, though @nathanielmanistaatgoogle may have a stronger opinion. 

> RE: 100% coverage. We need a "coalition of volunteers" to get #212 over the hill.

My time is a bit limited as I'm refactoring all of the tests in our sample repos, but happy to help with what's left after I'm done there.

**Comment by dhermes** [comment]
No worries, there should be plenty of stakeholders we could get to write one unit test?

**Comment by theacodes** [comment]
Thank you for creating this issue, however, this project is deprecatedand we will only be addressing critical security issues. You can read moreabout this deprecation [here](http://google-auth.readthedocs.io/en/latest/oauth2client-deprecation.html).

If you need support or help using this library, we recommend that you ask yourquestion on StackOverflow.

If you still think this issue is relevant and should be addressed, pleasecomment and let us know!
